src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TransPatterns.java

```diff
@@ -359,7 +356,7 @@ private MethodSymbol getAccessor(RecordComponent component) {
                                                   type,
                                                   currentClass);
             JCStatement accessorStatement =
-                    make.Return(make.App(make.Select(make.Ident(proxy.params().head), realAccessor)));
+                    make.Return(make.App(make.Select(make.Ident(proxy.params().head), c.accessor)));
```

Comment:
I let this to your consideration: it could be worth as part of this patch to check if there are other places in the record patterns code that can benefit from this simplification.

Reply:
Thanks, but I don't recall other places in the current pattern matching code that would be looking up the accessors.

Reply:
ok sounds good